cfpb/consumerfinance.gov
# Use TypedTableBlock instead of AtomicTableBlock

This PR adds a new Table block, based on Wagtail's [TypedTableBlock](https://docs.wagtail.org/en/v4.1.6/reference/contrib/typed_table_block.html) as an alternative to our existing AtomicTableBlock. See internal D&CP#254 for additional context for this work.

This PR adds the new Table option on all Page types and StreamField blocks where AtomicTableBlock is currently available, but does not (yet) remove ATB as an option.

I have written [a migration script](https://gist.github.com/chosak/4e1691685219f695b6824b32b0716903) that can be used to migrate all existing uses of ATBs to the new block. It is not 100% finished (for example, it depends on #5352) but is getting there. In the meantime, I'm opening this PR to allow people (primarily @csebianlander) to play around with the new Table block and give feedback before we migrate.

## Additions

New `v1.atomic_elements.tables.Table` block that implements the [Table component](https://cfpb.github.io/design-system/components/tables) from our Design System by leveraging Wagtail's TTB. The new Table block carries over only these two options from the existing ATB:

<img width="733" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/654645/2febd1bd-c0de-42c4-bc21-1037927f6a80">

This is based on a new `table/base.html` template that is greatly simplified from the ATB's template, which has been renamed to `table/atomic.html`. Notably, ATBs allow for display of raw, unescaped HTML, while the new table template will not allow this.

## Changes

This PR also migrates some of our existing custom table blocks to the new base template: the JobListingTable and ConsumerReportingCompanyTable.

When Wagtail editors go to add a block to a page, they'll now see "(Deprecated)" next to the existing AtomicTableBlock, for example when editing an Ask CFPB AnswerPage:

<img width="772" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/654645/04cd7953-51c8-4aac-a15a-f64185e20d51">

Once we migrate to this new block and deprecate ATBs completely, this duplication will go away.

## How to test this PR

Edit a page where you can currently add an ATB (for example, a DocumentDetailPage), and try adding a new Table instead.

There's a new template debug view for the new table template at http://localhost:8000/admin/template_debug/v1/table/.

To verify the JobListingTable changes, see the template debug view at http://localhost:8000/admin/template_debug/jobmanager/job_listing_table/ or visit http://localhost:8000/about-us/careers/current-openings/ with a production dump.

To verify the ConsumerReportingCompanyTable changes, see the template debug view at  http://localhost:8000/admin/template_debug/v1/crc_table/ or visit http://localhost:8000/consumer-tools/credit-reports-and-scores/consumer-reporting-companies/companies-list/a-plus-property-verisk/ with a production dump.

To experiment with migrating existing ATBs to TTBs, you can try [my migration script](https://gist.github.com/chosak/4e1691685219f695b6824b32b0716903):

```
% python migrate_atomic_tables.py your_wagtail_username --dry-run
```

(or without `--dry-run` to save and publish new revisions on all pages.) This script will only run against live pages in the main Wagtail site that currently have ATBs. It will replace those ATBs with an equivalent Table block of the new type. One example page to try this on is [this one](https://www.consumerfinance.gov/ask-cfpb/what-should-i-know-about-using-contractors-to-rebuild-after-a-disaster-en-1517/); you will see, however, that the script is still incomplete as it doesn't properly handle migrating rich text HTML yet.

## Screenshots

Adding a new table:

<img width="658" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/654645/17122009-356b-49ca-a208-183340962960">

Note that the table uses our standard Heading block, and supports columns of three types: text, numeric, and rich text.

## Notes and todos

TODO: Finish the migration script so it works properly on rich text content.

## Checklist
